### PR TITLE
Run runtime tests with mono on test change.

### PR DESCRIPTION
**Pull request by naricc**
Update runtime.yml so that we run the runtime tests using mono if tests are changes, or the build-test.sh script is changes (in addition to running them on coreclr).

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Whenever this is ready, can we please re-spin the CI couple of times before merging to identify flakiness and with that decrease the impact on the CI pass rate?

**Comment by naricc** [comment]
> Whenever this is ready, can we please re-spin the CI couple of times before merging to identify flakiness and with that decrease the impact on the CI pass rate?

Yes, happy to do that. If ti comes back with failures (for example, because runtime test were added before this change), I will probably need to add a change to issues.targets to this PR.

**Comment by safern** [comment]
This will also cause to run libraries tests against mono. Should we instead introduce a new subset in the paths like, runtimetests, and then modify the conditions appropriately?

**Comment by naricc** [comment]
> This will also cause to run libraries tests against mono. Should we instead introduce a new subset in the paths like, runtimetests, and then modify the conditions appropriately?

That is probably the better idea :)

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -695,6 +697,7 @@ jobs:
       condition: >-
         or(
           eq(dependencies.checkout.outputs['SetPathVars_mono.containsChange'], true),
+          eq(dependencies.checkout.outputs['SetPathVars_runtimetests.containsChange'], true),
```
Comment (safern) [inline eng/pipelines/runtime.yml:701]: You will also need to make sure mono is built (at least for OSX), same for CoreCLR Release, Libraries (Which I think already does).

Comment (naricc) [reply eng/pipelines/runtime.yml:701]: I added it for mono, it looks like CoreCLR and Libraries are always built, right? 
https://github.com/dotnet/runtime/blob/0fa0b905f65e57f5eb64dd556a07ead522930c8a/eng/pipelines/runtime.yml#L183
https://github.com/dotnet/runtime/blob/0fa0b905f65e57f5eb64dd556a07ead522930c8a/eng/pipelines/runtime.yml#L357

Comment (safern) [reply eng/pipelines/runtime.yml:701]: Yeah, those are always built.

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -293,6 +297,7 @@ jobs:
       condition: >-
         or(
           eq(dependencies.checkout.outputs['SetPathVars_libraries.containsChange'], true),
+          eq(dependencies.checkout.outputs['SetPathVars_runtimetests.containsChange'], true),
```
Comment (safern) [inline eng/pipelines/runtime.yml:300]: I guess building mono doesn't take long, but it still uses machines and cores and with all the core reduction we're trying to do, should we separate the OSX build into it's own entry? It seems overkill to build all of mono when runtime tests changes other than building only the mono matrix we need to run the runtime tests.

Comment (naricc) [reply eng/pipelines/runtime.yml:300]: I have another change I am working on to run these tests on more architectures. But maybe not all of them; I will make the change.

**Review by safern — APPROVED** [review]

File: eng/pipelines/runtime.yml [hunk 3]
```diff
@@ -296,6 +299,27 @@ jobs:
...
+    platforms:
+    - OSX_x64
+    jobParameters:
+      condition: >-
+        or(
+          eq(dependencies.checkout.outputs['SetPathVars_libraries.containsChange'], true),
+          eq(dependencies.checkout.outputs['SetPathVars_runtimetests.containsChange'], true),
+          eq(dependencies.checkout.outputs['SetPathVars_mono.containsChange'], true),
+          eq(variables['isFullMatrix'], true))
+
```
Comment (safern) [inline eng/pipelines/runtime.yml:320]: Nit: remove extra empty lines and just leave 1?

**Pull request merged**